### DISCUSSION TITLE
Does Copilot recommend code based on best practices?

### DISCUSSION THREAD

**Discussion by amponce** [community/community#6781] [Copilot Conversations]
So far I've been really impressed with the AI.  I'm curious how the recommendations align with popular patterns and best practices? Is it possible that old deprecated code could degrade the quality of the AI's recommendations? In other words does it adapt to the patterns in the existing repo or does it follow a certain guideline that is maintained and updated as technology and standards change?

**Comment by Urethramancer** [comment]
It seems to suggest based on most used code (and comments). I've had a few suggestions with misspellings, like "automaiton" and "conect". It seems to mostly fit my code though, but that could be because most of my code is Go, where the community agrees on interfaces and formatting.
